src/api/BaseServiceApi.ts

```diff
@@ -0,0 +1,83 @@
+import { API_PRODUCTS_URL } from '@/constants/envVariables'
```

Comment:
class BaseError extends Error {
constructor(mesage: string) {
    this.message = message
}
}

class BaseApi implements BaseApiInterface {
   constructor( baseUrl: string, error: Error = new BaseError('base error')){
       this.baseUrl = baseUrl
       this.error = error
   }
   
    private retriveData() {
       try {
          await fetch()
       } catch (e) {
          throw new this.error('network error')
        }
    }

   public get() {this.retriveData('get', params )}
   public post() {}
   ... delete, update
}